From jason.wessel@windriver.com  Tue May 12 09:47:08 2009
From: Jason Wessel <jason.wessel@windriver.com>
Date: Mon, 11 May 2009 15:24:08 -0500
Subject: USB: serial: ftd_sio usb: move status check
To: greg@kroah.com
Cc: stern@rowland.harvard.edu, Jason Wessel <jason.wessel@windriver.com>
Message-ID: <1242073450-3794-3-git-send-email-jason.wessel@windriver.com>


Alan Stern commented that the private driver counts must be updated
regard less of the status return on the urb when the write call back
is executed.

This patch alters the behavior to update the private driver counts by
simply moving the status check to after the driver count update.

Signed-off-by: Jason Wessel <jason.wessel@windriver.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/ftdi_sio.c |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

--- a/drivers/usb/serial/ftdi_sio.c
+++ b/drivers/usb/serial/ftdi_sio.c
@@ -1856,11 +1856,6 @@ static void ftdi_write_bulk_callback(str
 
 	dbg("%s - port %d", __func__, port->number);
 
-	if (status) {
-		dbg("nonzero write bulk status received: %d", status);
-		return;
-	}
-
 	priv = usb_get_serial_port_data(port);
 	if (!priv) {
 		dbg("%s - bad port private data pointer - exiting", __func__);
@@ -1878,6 +1873,11 @@ static void ftdi_write_bulk_callback(str
 	priv->tx_outstanding_bytes -= countback;
 	spin_unlock_irqrestore(&priv->tx_lock, flags);
 
+	if (status) {
+		dbg("nonzero write bulk status received: %d", status);
+		return;
+	}
+
 	usb_serial_port_softint(port);
 } /* ftdi_write_bulk_callback */
 
